src/controllers/user.controller.js

```diff
@@ -2,29 +2,77 @@ import { UserService } from '../services/user.service.js';
 import { saveTokens } from '../services/token.service.js';
 import { BcryptUtility } from '../utils/bcrypt.util.js';
 import { JwtUtility } from '../utils/jwt.util.js';
+import { sendEmail } from '../utils/sendEmail.js'
+import { MailConfigurations } from '../utils/mailConfigurations.js'
+import dotenv from 'dotenv';
+dotenv.config();
 
 export class UserController {
   static async registerUser(req, res) {
     try {
       const user = { ...req.body };
       user.password = await BcryptUtility.hashPassword(req.body.password);
-      const { id, email } = await UserService.register(user);
-      const userData = { id, email };
+      const { id, email, role } = await UserService.register(user);
+      const userData = { id, email, role };
       const userToken = JwtUtility.generateToken(userData);
       const data = {
         token: userToken,
         revoked: false,
       };
       await saveTokens.saveToken(data);
-      return res.status(201).json({ user: userData, token: userToken });
+	  if(process.env.NODE_ENV !== 'test'){
+	  	sendEmail(MailConfigurations.emailVerificationConfig(email, userToken));
+	  }
```

Comment:
I am using mock to test send email because sending test emails with a real transport can slow down tests and make them less reliable, especially if the email service is experiencing issues or network problems.